### DISCUSSION TITLE
Best way to make mobile version of site

### DISCUSSION THREAD

**Discussion by arso1er** [vercel/next.js#14162] [Help] [answered]
Hi,

So what do you think is the best way to make a site's mobile version knowing that the mobile version and the desktop version have little in common. I can think of two ways:
1. Use media queries, but the bundle size would be bigger which could degrade performance?
2. Have the mobile version in a subfolder (/mobile for example) and redirect each request accordingly. This seems the better way in my opinion, but the new getServerSideProps does not handle conditional redirect natively yet I think (would have to implement my own then).

What do you think? Do you know a better way to handle this?

Thanks.

**Comment by illuminist** [comment, marked as answer]
I prefer the 1.

- but the bundle size would be bigger which could degrade performance?

How much bigger? If 1-2 KB it's just negligible. Affected performance is only on first app load.

**Reply by arso1er** [reply]
The 1 looks like the better solution if bundle size does not increase too much. I guess I'll go with that first and if the bundle size turns out to be too big, I'll switch to 2.

Thanks for your answer.

**Reply by guiihlopes** [reply]
How to avoid the blinks using media queries? You must use a initial spinner always :/ ? And when you do want to make an adaptative layout? How to deal with the new getServerSideProps + getStaticProps? 

With getInitialProps we can check userAgent both in server and client to choose which component we want to show/hide.

**Reply by illuminist** [reply]
Not sure if that's a question. I think media query does not blink as browser will determine the style before painting DOM element. Another upside of using media query is you can basically use same component without need to determine user device, and the app will be able to adjust itself with narrow browser window for PC.

**Reply by guiihlopes** [reply]
In my past experiences using css-in-js it blinked. But the question is that I do want an adaptive layout, how to deal with the new getServerSideProps + getStaticProps?

**Comment by MohammadPCh** [comment]
It's not always about CSS.
Consider you need a swiper. In mobile devices we have swiper functionality with scroll. and we dont need third party libraries. but in desktop scrolls are ugly and we should use third party library like SwiperJS. and it increment our js bundle about 20kb!
